src/packages/core/protocol/__tests__/account.spec.ts

```diff
@@ -13,6 +14,52 @@ describe('create new accounts', () => {
  
 	describe('valid private key', () => {
 
+		describe('balance', () => {
+
+			it('getting the balance of an account', async () => {
+
+				// This API (explorer.nuls.services) doesn't currently work for new addresses
+				// If it would, this is what a test could look like:
+
+				// const account = Account.create();
+
+				// account.switchChain(ChainIdType.Testnet);
+				// let balance = await account.getBalance({host: 'https://explorer.nuls.services', base: ''});
+				// console.log(balance);
+
+				// expect(balance).toEqual({
+				// 	total: 0,
+				// 	locked: 0,
+				// 	usable: 0
+				// });
+				// Using a pre-defined address..
+
+				const resp = {total: 9999999999999, locked: 0, usable: 9999999999999};
+
+				//Account.getBalance = jest.fn((a, b) => {});
+				jest.spyOn(axios, 'get').mockImplementation(() => {
+					return Promise.resolve(resp);
+				});
+
+				const address = 'TTatyig2SCtmUEsgguKvxQQ421e6NULS';
+				const balance = await Account.getBalance(address, {host: 'https://explorer.nuls.services', base: ''});
+				const total = balance.total;
+				const locked = balance.locked;
+				const usable = balance.usable;
+
+				expect(locked + usable).toEqual(total);
+
+				expect(total).toBeGreaterThanOrEqual(0);
```

Comment:
Change all this `toBeGreaterThanOrEqual` and `toBeLessThanOrEqual` to three expects that check that the total, locked and usable returned values are exactly the same than you have mocked

Reply:
resolved